src/sagas/local-player.ts

```diff
@@ -108,23 +108,22 @@ function* handlePlaybackStateChange(
         return;
     }
 
-    if (spotifyState && spotifyState.track_window.current_track.id === currentTrack.reference.id) {
-        yield player.resume();
-    } else {
-        const playing = 'playing' in oldPlayback ? oldPlayback.playing : undefined;
-        const position = newPlayback.last_position_ms
-            ? newPlayback.last_position_ms + (playing !== false ? Date.now() - newPlayback.last_change : 0)
-            : 0;
+    const { currentParty, syncPlayback } = yield select((state: State) => {
+        return {
+            currentParty: state.party.currentParty,
+            syncPlayback: state.settingsView.syncedPlayback
+        }
+    })
 
-        yield all([
-            call(
-                playTrack,
-                currentTrack.reference.id,
-                deviceId,
-                position,
-            ),
-            call(markTrackAsPlayed, partyId, currentTrack.reference),
-        ]);
+    yield call(
+        playTrack,
+        currentTrack.reference.id,
+        deviceId,
+        syncPlayback ? currentParty.playback.last_position_ms : 0,
+    )
+
+    if (spotifyState && spotifyState.track_window.current_track.id === currentTrack.reference.id) {
+        yield call(markTrackAsPlayed, partyId, currentTrack.reference)
```

Comment:
@robhuzzey Were you able to test the changes in here? The playback code has to handle lots of edge cases (thats why it seems so badly written / unnecessarily complicated) and I'm not sure whether this isn't breaking some things. :)

(did't get the chance to test this yet, will do next week)

Reply:
I wasn't able to test it properly as I don't even have 2 machines to try it out on. I've asked a few people at work to give it a test with me when they have time....